core/envelope-sim/src/main/java/fr/sncf/osrd/envelope/Envelope.java

```diff
@@ -184,27 +185,48 @@ public double maxSpeedInRange(double beginPos, double endPos) {
         return maxSpeed;
     }
 
-    /** Computes the time required to get to a given point of the envelope */
-    public long interpolateTotalTimeUS(double position) {
+    private long interpolateUS(double position, boolean isArrivalAt) {
         assert continuous : "interpolating times on a non continuous envelope is a risky business";
         var envelopePartIndex = findLeft(position);
+        if (!isArrivalAt) envelopePartIndex = findRight(position);
```

Comment:
In this context I don't think we need to make a difference, speed can have discontinuities across parts but time should be continuous. Stops are instant in this context.

Reply:
Removed in https://github.com/OpenRailAssociation/osrd/pull/7609/commits/b7f044ebd08664f02948f7aafb19982fee469f65